src/daemon/mod.rs

```diff
@@ -393,21 +391,30 @@ pub(super) async fn start(
         &config.client.data_dir,
     );
 
+    // Sets the latest snapshot if needed for downloading later
     let mut config = config;
-    fetch_snapshot_if_required(&mut config, epoch, opts.auto_download_snapshot).await?;
-
-    if let Some(path) = &config.client.snapshot_path {
-        let stopwatch = time::Instant::now();
-        import_chain(
-            &state_manager,
-            &path.display().to_string(),
-            config.client.skip_load,
-            config.client.chunk_size,
-            config.client.buffer_size,
+    if config.client.snapshot_path.is_none() {
+        set_snapshot_path_if_needed(
+            &mut config,
+            epoch,
+            opts.auto_download_snapshot,
+            &db_root_dir,
         )
-        .await
-        .context("Failed miserably while importing chain from snapshot")?;
-        info!("Imported snapshot in: {}s", stopwatch.elapsed().as_secs());
+        .await?;
+    }
+
+    // Import chain if needed
+    if !opts.skip_load.unwrap_or_default() {
+        if let Some(path) = &config.client.snapshot_path {
+            // TODO: respect `--consume-snapshot` CLI option once it's implemented
```

Comment:
do we have a tracking issue for this?

Reply:
Yes it's part of https://github.com/ChainSafe/forest/issues/3334

Reply:
let's have the link in the comment

Reply:
Done